wal/wal_test.go

```diff
@@ -547,6 +548,125 @@ func TestWAL_Repair(t *testing.T) {
 	}
 }
 
+func TestCorruptAndCarryOn(t *testing.T) {
```

Comment:
The name is quite clear, but if you don't mind please add a more extended comment the behaviour that this test is verifying. It will probably be better when looking at it in few months time. For example if some code changes cause it to fail.

Reply:
I meant something like

```
TestCorruptAndCarryOn ensures that an error during reading the segment .... and moves to reading the next segment only after .....
```